x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/all/index.tsx

```diff
@@ -318,24 +364,47 @@ export const AllRules = React.memo<AllRulesProps>(
                     </UtilityBarGroup>
                   </UtilityBarSection>
                 </UtilityBar>
-                <MyEuiBasicTable
-                  data-test-subj="rules-table"
-                  columns={columns}
-                  isSelectable={!hasNoPermissions ?? false}
-                  itemId="id"
-                  items={rules ?? []}
-                  noItemsMessage={emptyPrompt}
-                  onChange={tableOnChangeCallback}
-                  pagination={{
-                    pageIndex: pagination.page - 1,
-                    pageSize: pagination.perPage,
-                    totalItemCount: pagination.total,
-                    pageSizeOptions: [5, 10, 20, 50, 100, 200, 300],
-                  }}
-                  ref={tableRef}
-                  sorting={{ sort: { field: 'enabled', direction: filterOptions.sortOrder } }}
-                  selection={hasNoPermissions ? undefined : euiBasicTableSelectionProps}
-                />
+                {tabs}
+                {allRulesTab === AllRulesTabs.rules && (
+                  <MyEuiBasicTable
+                    data-test-subj="rules-table"
+                    columns={columns}
+                    isSelectable={!hasNoPermissions ?? false}
+                    itemId="id"
+                    items={rules ?? []}
+                    noItemsMessage={emptyPrompt}
+                    onChange={tableOnChangeCallback}
+                    pagination={{
+                      pageIndex: pagination.page - 1,
+                      pageSize: pagination.perPage,
+                      totalItemCount: pagination.total,
+                      pageSizeOptions: [5, 10, 20, 50, 100, 200, 300],
+                    }}
+                    ref={tableRef}
+                    sorting={{ sort: { field: 'enabled', direction: filterOptions.sortOrder } }}
+                    selection={hasNoPermissions ? undefined : euiBasicTableSelectionProps}
+                  />
+                )}
+                {allRulesTab === AllRulesTabs.monitoring && (
+                  <MyEuiBasicTable
+                    data-test-subj="monitoring-table"
+                    columns={getMonitoringColumns()}
+                    isSelectable={!hasNoPermissions ?? false}
+                    itemId="id"
+                    items={rulesStatuses}
+                    noItemsMessage={emptyPrompt}
+                    onChange={tableOnChangeCallback}
+                    pagination={{
+                      pageIndex: pagination.page - 1,
+                      pageSize: pagination.perPage,
+                      totalItemCount: pagination.total,
+                      pageSizeOptions: [5, 10, 20, 50, 100, 200, 300],
+                    }}
+                    ref={tableRef}
+                    sorting={{ sort: { field: 'enabled', direction: filterOptions.sortOrder } }}
```

Comment:
try to avoid passing object inline as a prop, because it will cause the component to re-render

Reply:
I'm confused. From what I'm reading [1] I'm not sure if `useRef` does re-render. The docs say if the `.current` property is modified, which it is in the reducer, it should not cause the component to re-render. Let me know if I'm misunderstanding the problem in this instance, which I quite possibly am.

[1]: https://reactjs.org/docs/hooks-reference.html#useref

Reply:
@dhurley14 I did mean `sorting` field, sorry for being unclear ;(

Reply:
Ah! okay gotcha. Sorry about that.